kodadot/nft-gallery
# Multi currency implementation for Basilisk on Rococo in KodaDot

# Please test everything again

## What's new
- route /snek
- you should see KSM there
- pay for fees in KSM / BSX
- for everything else use KSM
- faucet soon

## What to test:

- Create collection
- Create NFT
- change your fee payment asset to `KSM` (`/asset`)
- In balanceInputs you should see `KSM` otherwise report as bug
- List NFT
- Make Offer
- MultiFee should always tell you correct currency u use ('You are paying fees in {currency}')
- New prefix is `/snek`

## Things I am aware that does not work correctly

- [ ] New balanceinput has strange validation (always red even if the input is correct)
- [ ] Most probably I broke `/bsx` (we will sunset)
- [ ] Format balance is not the same as Money

## 🐍 changelog

- :snake: snake config
- :zap: added snek
- :bug: max royalty value is 99%
- :spaghetti: pasta from bsx
- :snake: added snek to utils
- :zap: query utils for basilisk
- :zap: Button tells you in which currency u pay for fees
- :label: import Codec as type
- :globe_with_meridians: paying fees in has translation
- :zap: create token has cute link to assets
- :zap: utility package for bsx specific calls
- :alien: asset list for bsx
- :tada: asset page
- :construction: Asset Page
- :globe_with_meridians: asset page
- :bug: :label: AssetItem has id
- :zap: AssetList can fetch user
- :globe_with_meridians: better wording for assets
- :construction: Reworking identity store
- :zap: action to fetch balance
- :bug: correctly fetch balance #3587
- :test_tube: :zap: added data-testid
- :test_tube: added test for BasicBalanceInput
- :alien: asset id for bsx
- :zap: Input for Special tokens
- :zap: CreateToken is using KSM
- :tada: BasicMoney is here
- :zap: ChainMoney as the Money
- :bug: BasicMoney has bad filter
- :bug: ChainMoney should send value
- :zap: show balance of the particular token id
- :bug: Invalid state is handled as 0
- :zap: tokenBalanceInput
- :zap: bsx use multicurrency impl
- :bug: Debounce caused a ton of bugs
- :snake: recognize snek as chain with actions
- :mute: Deleted not fancy log
- :zap: Available actions have dynamic props :)
- :construction_worker: account balance will also handle tokens
- :tada: asset store is here
- :zap: useAppolo can reuse also the apolloClient
- :construction_worker: subscribe for token balance
- :construction_worker: :bug: handle if chain has no token pallet
- :bug: autofetching token balance :white_check_mark:
- :zap: function for gettingTokenBalance
- :construction_worker: WIP OfferBalanceInput
- :zap: asset mixin is here
- :top: export TokenProperty
- :construction_worker: WIP TokenBalanceInput is Using store
- :package: @kodadot1/vuex-options@0.0.1-rc.11
- :broom: remove unused functions

**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).
👇 \_ Let's make a quick check before the contribution.

### PR type


### What's new?

- [ ] Requires deployment <rubick/magick_issue>

### Before submitting Pull Request, please make sure:


### Optional
